source/common/common/thread.h

```diff
@@ -183,12 +200,26 @@ struct MainThread {
       return true;
     }
     // When threading is on, compare thread id with main thread id.
-    return MainThreadSingleton::get().inMainThread();
+    return MainThreadSingleton::get().inMainThread() || MainThreadSingleton::get().inTestThread();
```

Comment:
this is a bit of a hack in prod code. If we used the ThreadFactory abstraction per above, we could easily make a TestThreadFactory that added this functionality rather than having prod code call test-specific code.

How about a TODO for now?

Reply:
Thanks, will do.